### PR TITLE
batch request for tasks 7 and 8 due to branching issue

### PR DESCRIPTION
**Pull request by lai1372**

### CODE REVIEW

File: src/components/UpVotes.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    patchReviewDownVote(review_id).catch(() => {
+      setVotes((currentVotes) => {
+        return currentVotes + 1;
+      });
+    });
+    setThumbsDown(true)
+    setThumbsUp(false)
+  };
+
+  const clearVotes = () =>{
```
Comment (JVal-NC) [inline src/components/UpVotes.jsx:38]: I like what you've done but could something more typical of UI seen on other websites be replicated?

File: src/components/DownVotes.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+import { patchReview } from "../../utils.js";
+
+const DownVotes = ({ review_id, review_votes }) => {
```
Comment (JVal-NC) [inline src/components/DownVotes.jsx:4]: Is this component being used at all? If not lets get rid of it.

File: src/components/Comments.jsx [hunk 3]
```diff
@@ -1,35 +1,41 @@
...
         setIsLoading(false);
       });
-  }, [comments]);
+  }, []);
 
   if (isLoading) {
     return <p>Loading...</p>;
   }
+
+  if (!comments.length) {
```
Comment (JVal-NC) [inline src/components/Comments.jsx:22]: Missing an AddComment component in this scenario.

File: src/components/Comments.jsx [hunk 4]
```diff
@@ -1,35 +1,41 @@
...
-          <p>Votes: {comment.votes}</p>
-          <p>Post date: {comment.created_at}</p>
-        </p>
+      {comments.map((comment) => {
+        return (
+          <article key={comment.comment_id} className="single-comment">
+            <p>Written by: {comment.author}</p>
+            <p>Comment: {comment.body}</p>
+            <p>Votes: {comment.votes}</p>
+            <p>Post date: {comment.created_at}</p>
```
Comment (JVal-NC) [inline src/components/Comments.jsx:36]: Can we format this date to something more readable.

File: src/components/AddComment.jsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import { useState } from "react";
+import { fetchUsers, postComment } from "../../utils";
+
+const AddComment = ({ review_id, setComments }) => {
+  const [newUsername, setNewUsername] = useState("grumpy19");
+  const [newComment, setNewComment] = useState("");
+  const [errorMsgWhiteSpace, setErrorMsgWhiteSpace] = useState(null);
+  const [errorMsgCharacter, setErrorMsgCharacter] = useState(null);
+  const [characterLength, setCharacterLength] = useState(0);
```
Comment (JVal-NC) [inline src/components/AddComment.jsx:9]: Is this state being used?

File: src/components/AddComment.jsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+          return [response.data.comment, ...currentComments];
+        });
+      })
+      .catch((err) => console.log(err));
+
+    setNewUsername("");
+    setNewComment("");
+  };
+
+  const handleCommentValidation = (event) => {
```
Comment (JVal-NC) [inline src/components/AddComment.jsx:31]: Good work so far, what about an empty comment? Seems like the newUsername gets reset to blank when trying to post a null comment.

File: src/components/Comments.jsx [hunk 7]
```diff
@@ -1,35 +1,41 @@
...
+      <AddComment review_id={review_id} setComments={setComments} />
       <h2>Comments</h2>
-      {comments.map((comment)=>{
-        return <p className="single-comment">
-          <p>Written by: {comment.author}</p>
-          <p>Comment: {comment.body}</p>
-          <p>Votes: {comment.votes}</p>
-          <p>Post date: {comment.created_at}</p>
-        </p>
+      {comments.map((comment) => {
```
Comment (JVal-NC) [inline src/components/Comments.jsx:30]: For comments over a certain number of characters the text overflows, see route below:

reviews/13

**Review by mallionaire — APPROVED** [review]
Nice looking app for the most part, however the background colour and the text on your comments section don't have enough contrast for them to be readable - if this is placeholder styling for the moment please choose a lighter background :smile: 

I'd also like you to think about your commit messages, there should be frequent commits, and the messaged should clearly describe the functionality that you jave added - they should never refer to "task X" - it will be very hard to remember what "task X" was if you look at this repo in a few months time.

**Pull request merged**